scourgify/address_constants.py

```diff
@@ -16,6 +16,13 @@
     'CT': 'COURT'
 }
 
+AMBIGUOUS_DIRECTIONALS = {
+    'NORTH-WEST': 'NW',
```

Comment:
No.  Hyphens are a valid address elements in a number of places, just not in these directional typos.  So you can't clear the hyphens out before parsing the address, and if you parse the address before clearing these typos up, then the directionals will get miss-tagged.

Reply:
I still think it's possible to combine this with the other directional lookups, rather than having it separate. Here is some quasi C# (since I can think better in that language):

```csharp
// object declaration to hold each directional with all it's variations.
public class Directional
{
    public string Abbreviation { get; set; }
    public string LongHand { get; set; }
    public string[] Variations { get; set; }
}

// example of defining the lookup
Directional[] directionals = [
    new Directional 
    {
        Abbreviation = "NW",
        LongHand = "NORTHWEST",
        Variations = [ "NORTHWEST", "NORTH-WEST", "N.W." ]
    }
];

// example lookup method
Directional ParseDirectional(string input)
{
    return directionals.First(d => 
        d.Abbreviation == input ||
        d.LongHand == input ||
        d.Variations.Contains(input));
}
```

and here I ran that through an AI converter to python (not sure how accurate it is:
```python
class Directional:
    def __init__(self, abbreviation, long_hand, variations):
        self.abbreviation = abbreviation
        self.long_hand = long_hand
        self.variations = variations

directionals = [
    Directional("NW", "NORTHWEST", ["NORTHWEST", "NORTH-WEST", "N.W."])
]

def parse_directional(input):
    return next((d for d in directionals if input in [d.abbreviation, d.long_hand] or input in d.variations), None)
```

Reply:
The problem with doing it this way is that the package does not just take in an address and parse it as though it were a string.
Let's say I have an address: `123 NorthWest West Str Apt S'. 
If I just parse this string to standardize directionals and street types, and such, I'm going to end up with either:
`123 NW W ST APT S', if I request the default output.  Or `123 NORTHWEST WEST STREET APT SOUTH` if I requested the long_hand output.
Instead, this package relies on `usaddress`.  When I call `ussaddress.tag` on the above address I get:
```
(OrderedDict([('AddressNumber', '123'),
              ('StreetNamePreDirectional', 'NorthWest'),
              ('StreetName', 'West'),
              ('StreetNamePostType', 'Str'),
              ('OccupancyType', 'Apt'),
              ('OccupancyIdentifier', 'S')]),
 'Street Address')
 ```
 This way I can standardize only the select parts that should be standardized, pre or post directional, street type (but not street name), and occupancy type (but not occupancy identifier).
 But if I write the address with a hyphenated directional, the tagging does not correctly identify the directional as a directional:
 ```
 (OrderedDict([('AddressNumber', '123'),
              ('StreetName', 'North-West West'),
              ('StreetNamePostType', 'Str'),
              ('OccupancyType', 'Apt'),
              ('OccupancyIdentifier', 'S')]),
 'Street Address')
 ```
 That is why those, and those alone, have to be preparsed in the address string as part of the pre-cleaning process.  In addition to this cleanup, the precleaning process removes extra spaces and most, if not all, punctuation, other than hyphens.  Hyphens can be a valid part of a house number or street name, so those can't be stripped out.  Which is why the only way to cleanup hyphenated directionals is the find and replace.  But again, I can't do a find and replace for all possible directional typos, abbreviations, or variants in the address string, because a directional street name is a perfectly valid street name that should not be abbreviated.